dguta/db.go

```diff
@@ -510,9 +513,22 @@ func (d *DB) storeDGUTAs(tx *bolt.Tx) error {
 // storeDGUTA stores a DGUTA in the db. DGUTAs are expected to be unique per
 // Store() operation and database.
 func (d *DB) storeDGUTA(b *bolt.Bucket, dguta *DGUTA) error {
-	dir, gutas := dguta.encodeToBytes(d.ch)
+	var dgutas [len(summary.DirGUTAges)]DGUTA
+
+	for _, v := range dguta.GUTAs {
+		dgutas[v.Age].GUTAs = append(dgutas[v.Age].GUTAs, v)
+	}
+
+	for age, v := range dgutas {
+		v.Dir = string(byte(age)) + dguta.Dir
+		dir, gutas := v.encodeToBytes(d.ch)
+
+		if err := b.Put(dir, gutas); err != nil {
```

Comment:
You'll need to ensure that you put items in the db in key sorted order, or this operation will be unusable slow.